tests/code_size/hello_world_wasm2js.js

```diff
@@ -0,0 +1,73 @@
+var b = Module;
+
+var d = new TextDecoder("utf8");
+
+function e(a) {
+    if (!a) return "";
+    for (var l = a + NaN, c = a; !(c >= l) && f[c]; ) ++c;
+    return d.decode(f.subarray(a, c));
+}
+
+var g = new function(a) {
+    this.buffer = new ArrayBuffer(65536 * a.initial);
+}({
+    initial: 256,
+    maximum: 256
+}), f = new Uint8Array(g.buffer);
+
+f.set(new Uint8Array(b.mem), 1024);
+
+var h = {
+    a: function(a) {
+        console.log(e(a));
+    },
+    memory: g
+}, k, m = (new function() {
+    this.exports = function instantiate(t) {
+        function r(u) {
+            u.set = function(v, w) {
+                this[v] = w;
+            };
+            u.get = function(v) {
+                return this[v];
+            };
+            return u;
+        }
+        function s(x) {
+            var a = Math.imul;
+            var b = Math.fround;
+            var c = Math.abs;
+            var d = Math.clz32;
+            var e = Math.min;
+            var f = Math.max;
+            var g = Math.floor;
+            var h = Math.ceil;
+            var i = Math.trunc;
+            var j = Math.sqrt;
+            var k = x.abort;
+            var l = NaN;
+            var m = Infinity;
```

Comment:
Yikes - This looks like something that could be low hanging to DCE? I wonder why Closure did not clean them.. :/

Reply:
@alon...   What phase would we expect to clean this up?  Is it something that closure should be able deal with?

Reply:
I believe this is inside the wasm2js output. The JS optimizer doesn't normally run on that. We usually don't run a JS minifier on compiled code, as it could be massive. But you can try `--closure 2` which I think does that.

An option is to handle this in wasm2js, that is, track what is needed and emit only that. Not sure how easy it would be.

Reply:
But that about closure compiler itself.. .shouldn't it be able to eliminate these unused variables?

Reply:
Closure would be able to do it., but closure does not run on the wasm2js output - except for `--closure 2` (because, as I said, the compiled code can be massive).

Reply:
Ah I see.  Thanks.   I wonder if we should be running with `--closure 2` for these minimal codesize tests?  @juj?

Reply:
Because it falls down on huge files, I don't think we recommend `--closure 2`. But it would be good to investigate that, as maybe closure has gotten faster/uses less memory now?